### PR TITLE
fix(controller): Fix node status when daemon pod deleted but its children nodes are still running

### PR DESCRIPTION
**Pull request by lonsdale8734**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/controller/dag.go [hunk 1]
```diff
@@ -691,6 +691,11 @@ type TaskResults struct {
 // evaluateDependsLogic returns whether a node should execute and proceed. proceed means that all of its dependencies are
 // completed and execute means that given the results of its dependencies, this node should execute.
 func (d *dagContext) evaluateDependsLogic(taskName string) (bool, bool, error) {
+	node := d.getTaskNode(taskName)
+	if node != nil {
+		return true, true, nil
+	}
```
Comment (simster7) [inline workflow/controller/dag.go:697]: Can you explain this particular change more? Not sure if we want this

File: workflow/controller/operator.go [hunk 2]
```diff
@@ -917,6 +917,7 @@ func (woc *wfOperationCtx) podReconciliation() error {
 
 			node.Message = "pod deleted"
 			node.Phase = wfv1.NodeError
+			node.Daemoned = nil
```
Comment (alexec) [inline workflow/controller/operator.go:920]: what is this needed for please?

File: workflow/controller/operator.go [hunk 3]
```diff
@@ -1069,6 +1070,7 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 		if pod.DeletionTimestamp != nil {
 			// pod is being terminated
 			newPhase = wfv1.NodeError
+			newDaemonStatus = pointer.BoolPtr(false)
```
Comment (alexec) [inline workflow/controller/operator.go:1073]: and the purpose of this?

File: workflow/controller/operator.go (hunk 2 above)
Comment (lonsdale8734) [reply workflow/controller/operator.go:920]: When daemon pod deleted and `Daemoned` flag in node is true, there is a bug that the `Deaemoned` flag never removed and the whole workflow will never completed.

File: workflow/controller/operator.go (hunk 3 above)
Comment (lonsdale8734) [reply workflow/controller/operator.go:1073]: Ditto.

**Review by simster7 — APPROVED** [review]

**Comment by simster7** [comment]
This fix is out on https://github.com/argoproj/argo/releases/tag/v2.12.5

**Pull request merged**